src/ng/directive/ngClass.js

```diff
@@ -8,92 +8,93 @@
 
 function classDirective(name, selector) {
   name = 'ngClass' + name;
-  return ['$animate', function($animate) {
+  return ['$animate','$parse', function($animate,$parse) {
     return {
       restrict: 'AC',
-      link: function(scope, element, attr) {
-        var oldVal;
-
-        scope.$watch(attr[name], ngClassWatchAction, true);
-
-        attr.$observe('class', function(value) {
-          ngClassWatchAction(scope.$eval(attr[name]));
+      compile: function(tElement, tAttr) {
+        var classFn = $parse(tAttr[name], function(value) {
```

Comment:
The overhead should be negligible (and not enough to justify the breaking change).

<sub>(Note that `$parse` has a cache, so it doesn't have to actually lex/parse the same expressiom twice. It only has to retrieve the parsed expression and apply the interceptor.)</sub>